### ISSUE TITLE
SVM Data set , classifier except knn

### ISSUE THREAD

**Issue by maral-anahid** [maral-anahid/machine-learning-#2]
step-1

from sklearn.datasets import load_iris
import pandas as pd

# Load dataset
iris = load_iris()
X = pd.DataFrame(iris.data, columns=iris.feature_names)
y = pd.Series(iris.target)

# Dataset info
print("Shape:", X.shape)
print("Classes:", iris.target_names)
print("\nDataset Summary:")
print(X.describe())

این بخش داده معروف Iris را از کتابخانه scikit-learn بارگذاری می‌کند، سپس اطلاعات آماری و ساختار داده را نمایش می‌دهد.
داده شامل 150 نمونه و 4 ویژگی است و سه کلاس گل دارد: setosa, versicolor, virginica.

----------------------------------
step-2

import numpy as np
import matplotlib.pyplot as plt
from sklearn import svm

# Use two features for 2D visualization
X_plot = X[['petal length (cm)', 'petal width (cm)']]
y_plot = y

# Train SVM with linear kernel
clf = svm.SVC(kernel='linear', C=1.0)
clf.fit(X_plot, y_plot)

# Create meshgrid for decision boundary
xx, yy = np.meshgrid(np.linspace(1, 7, 200), np.linspace(0, 3, 200))
Z = clf.predict(np.c_[xx.ravel(), yy.ravel()])
Z = Z.reshape(xx.shape)

# Plot
plt.figure(figsize=(8,6))
colors = ['red', 'green', 'blue']
for i, color in enumerate(colors):
    plt.scatter(X_plot.iloc[y_plot==i, 0], X_plot.iloc[y_plot==i, 1],
                label=iris.target_names[i], color=color)

plt.contourf(xx, yy, Z, alpha=0.25)
plt.xlabel('Petal length (cm)')
plt.ylabel('Petal width (cm)')
plt.legend()
plt.title('SVM Decision Boundary (Linear Kernel)')
plt.show()

در این بخش از دو ویژگی مهم (طول و عرض گلبرگ) استفاده شده است تا مدل SVM خطی آموزش ببیند و خط جداکننده بین کلاس‌ها ترسیم شود.
نتیجه نشان می‌دهد که Setosa به‌طور کامل از دو کلاس دیگر جدا می‌شود و بین Versicolor و Virginica یک مرز خطی قابل‌قبول وجود دارد.

----------------------------------
step-3

from sklearn.model_selection import train_test_split
from sklearn.metrics import accuracy_score, confusion_matrix, classification_report
from sklearn.linear_model import LogisticRegression
from sklearn.tree import DecisionTreeClassifier
from sklearn.ensemble import RandomForestClassifier

# Split dataset
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)

# Define models
models = {
    "SVM (Linear)": svm.SVC(kernel='linear'),
    "Logistic Regression": LogisticRegression(max_iter=1000),
    "Decision Tree": DecisionTreeClassifier(),
    "Random Forest": RandomForestClassifier()
}

# Train and evaluate
results = {}
for name, model in models.items():
    model.fit(X_train, y_train)
    y_pred = model.predict(X_test)
    acc = accuracy_score(y_test, y_pred)
    results[name] = acc
    print(f"\n===== {name} =====")
    print("Accuracy:", round(acc, 3))
    print("Confusion Matrix:\n", confusion_matrix(y_test, y_pred))
    print("Classification Report:\n", classification_report(y_test, y_pred, target_names=iris.target_names))

----------------------------------
step-4

import matplotlib.pyplot as plt

plt.figure(figsize=(6,4))
plt.barh(list(results.keys()), list(results.values()), color='teal')
plt.xlabel("Accuracy")
plt.title("Model Accuracy Comparison")
plt.xlim(0.9, 1.0)
plt.show()

<img width="1580" height="980" alt="Image" src="https://github.com/user-attachments/assets/eb81a38d-e274-4333-aae0-fe891f2bf10f" />

----------------------------------
step-5

 از بین مدل‌های بررسی‌شده، SVM با کرنل خطی بهترین عملکرد را داشته است.
داده‌ی Iris نشان می‌دهد که تنها با دو ویژگی «طول گلبرگ» و «عرض گلبرگ» می‌توان گونه‌های مختلف را تقریباً با دقت کامل از هم تفکیک کرد.
این تمرین اهمیت انتخاب ویژگی مناسب و مقایسه مدل‌های مختلف را در یادگیری ماشین نشان می‌دهد.
